### ISSUE TITLE
[QoL]Add a !online or !timeonline command, allowing for your total playtime for your current session to be broadcasted to other runelite users

### ISSUE THREAD

**Issue by Shaidiuadih** [runelite/runelite#11037]
Hello!

I'm sure a lot of players who use runelite also have pures, or other builds that they are frequently creating and spending many hours training to perfection. I know I do!

I'm asking if it's possible to create a command that would show your total playtime for your current session to other runelite users that would help you combat situations where you are going to either reset your aggression timers, or bank, and coming back to someone being in your spot, who have recently or just logged into your world.

Jagex changed how drops function a while back, and if you log into a world, you will not be able to see any drops for x amount of minutes before having been logged into said world, so a lot of people who log into worlds are having a hard time believing people who claim to have been there for many hours already, and this causes a lot of conflicts and arguments between players that is unnecessary and shouldn't be taking place to begin with.

I feel like this is very much a quality of life update, and I am sure a lot of people would appreciate an update like this coming to the client. The client is already great, and has many awesome features readily available, and that's awesome!

Thank you for reading this.

P.S.

I realize that you can't really hold jeopardy on a spot, or a world at all, but we have played this game for many years together, and there has always been a mutual respect for one another, and I think that adding this feature would just add another layer to that.

**Comment by Nightfirecat** [comment]
I don't think this would be useful as players could spoof that information very easily.

**Comment by deathbeam** [comment]
I like this, its innocent enough and people can spoof our existing stuff already anyway.

**Comment by Shaidiuadih** [comment]
> I don't think this would be useful as players could spoof that information very easily.

This would easily be avoidable by simply adding a colour that doesn't exist in the game to any text broadcasted using this command. Honestly baffled you hadn't thought of this.

> I like this, its innocent enough and people can spoof our existing stuff already anyway.

Sure, they could try to do this, but it would be, as the aforementioned states, be avoidable by simply adding a colour to the next shown when using this command.

**Comment by Adam-** [comment]
> 
> 
> > I don't think this would be useful as players could spoof that information very easily.
> 
> This would easily be avoidable by simply adding a colour that doesn't exist in the game to any text broadcasted using this command. Honestly baffled you hadn't thought of this.
> 
> > I like this, its innocent enough and people can spoof our existing stuff already anyway.
> 
> Sure, they could try to do this, but it would be, as the aforementioned states, be avoidable by simply adding a colour to the next shown when using this command.

"Spoofing" means sending forged http requests to the server which shares the data between clients, not people typing out the text themselves. So using "color" would not fix this at all.

That said, we have a number of spoofable things already like !kc. I think as long as there is no financial incentive for people to spoof things it is mostly acceptable to have, as it limits it to just a handful of isolated people doing it instead of having organized groups of people trying to do it.

**Comment by Shaidiuadih** [comment]
> > > I don't think this would be useful as players could spoof that information very easily.
> > 
> > 
> > This would easily be avoidable by simply adding a colour that doesn't exist in the game to any text broadcasted using this command. Honestly baffled you hadn't thought of this.
> > > I like this, its innocent enough and people can spoof our existing stuff already anyway.
> > 
> > 
> > Sure, they could try to do this, but it would be, as the aforementioned states, be avoidable by simply adding a colour to the next shown when using this command.
> 
> "Spoofing" means sending forged http requests to the server which shares the data between clients, not people typing out the text themselves. So using "color" would not fix this at all.
> 
> That said, we have a number of spoofable things already like !kc. I think as long as there is no financial incentive for people to spoof things it is mostly acceptable to have, as it limits it to just a handful of isolated people doing it instead of having organized groups of people trying to do it.

Ah, yes. I've seen people fake kill counts before, and to my knowledge, this is going into the files, and editing the contents. I'm not really savvy when it comes to coding, so I wouldn't know where to look; but wouldn't this just result in a static time that doesn't change (unless they do it themselves, every time they wish to use the command) when you use the command? A person using this, would use !online, or !timeonline and it would say [03:43:57] for example, twice, when using the command, even if the command is executed with an interval of 20 seconds or more?

Just going by what I've been told by other players, and I'm just submitting this to help people who are having troubles finding worlds due to being low levels and targeted as a result thereof.

Thanks for your reply!

**Comment by Matsyir** [comment]
> wouldn't this just result in a static time that doesn't change (unless they do it themselves, every time they wish to use the command) when you use the command? A person using this, would use !online, or !timeonline and it would say [03:43:57] for example, twice, when using the command, even if the command is executed with an interval of 20 seconds or more?

It could be that way if it was done poorly, but one could also just use the client code/developer tools and modify the plugin to make it send the real time + 2 hours or whatever they want instead of the intended value.

Regardless, I agree with deathbeam, this is harmless so sure why not. But I don't think players in the examples you describe would change their actions because of this command.

**Comment by Shaidiuadih** [comment]
> It could be that way if it was done poorly, but one could also just use the client code/developer tools and modify the plugin to make it send the real time + 2 hours or whatever they want instead of the intended value.
> 
> Regardless, I agree with deathbeam, this is harmless so sure why not. But I wouldn't expect players in the examples you describe would change their actions because of this command.

I see!

I was not aware that this was a thing... My bad. Like I said, I'm not very savvy when it comes to coding, and I had no idea that developer mode is something available to anyone and everyone.

To adress your concern though, this was mostly directed at situations where you accidentally end up crashing somebody, because there is a confusion between the two players.

You run to the bank, to restock on supplies, and come back to find that somebody has taken your spot, so you could nicely ask them to leave, and they would often times be hesitant, because you have no way of proving otherwise; with this command, I hope that more people are able to retain their training spots, or any other spot for activity by quoting their total playtime for their session, in hopes of the other player recognizing this, and leaving. The same would apply to running to reset the aggression timers that have expired.

I agree though that anyone could just spoof this, and use it to trick others, but would it be possible to not only have this command display your total playtime for this session, but also any experience and in what skill you've gained as well?

This game is very competitive, and making builds for activities such as PKing, or bossing/revenants has never been more relevant than now, so it is rough finding a world to train your account in. Though, most people aren't dicks to one another, and would simply hop if the opposing player has the grounds to back up their claims.

I am just trying to be nice here, but I see a lot of people fall victim to this and it sucks to see, because this is a game we all know and love. I wish to see it prosper for a lot longer, because it is a great game. I grew up playing this game, so it is only natural for me to feel this sentimental towards it, and its players.

If this suggestion lacks grounds, or enough support to warrant an implementation, there is always tomorrow. I do hope that this garners enough attention though, to make it into the live game.

Thank you for your reply, and your clarification.

**Comment by Matsyir** [comment]
>you could nicely ask them to leave, and they would often times be hesitant, because you have no way of proving otherwise (...) most people aren't dicks to one another, and would simply hop if the opposing player has the grounds to back up their claims.

I agree most people aren't as toxic as a lot of people say. And I can relate to the situations you're describing, I've trained many pures at sand crabs. I just think that those who would be willing to leave with this timer, already would have been willing to leave previously. There already kind of is some level of 'proof' that you were already there in this situation: you arrive to that spot very shortly after the other player got there. If that wasn't enough for them to believe you and hop, I think it would be rare that this timer makes a difference. You could also just have been logged into that world doing something else. Again though: nothing against the suggestion, just wanted to clarify these concerns in relation to your proposed use cases.
